ReportSimulationOutput/measure.rb

```diff
@@ -1715,6 +1831,27 @@ def get_report_variable_data_annual(key_values, variables, unit_conv = UnitConve
     return val
   end
 
+  def get_resilience_timeseries(init_time_step, batt_kw, batt_soc_kwh, crit_load, n_timesteps, ts_per_hr)
+    (0...n_timesteps).each do |i|
+      t = (init_time_step + i) % n_timesteps # for wrapping around end of year
+      load_kw = crit_load[t]
+
+      if load_kw > 0
+        if [batt_kw, batt_soc_kwh].min >= load_kw # battery can carry balance
+          # prevent battery charge from going negative
+          batt_soc_kwh = [0, batt_soc_kwh - load_kw].max
+          load_kw = 0
+        end
+      end
+
+      if load_kw > 0 # failed to meet load in this time step
+        return i / Float(ts_per_hr)
```

Comment:
@jmaguire1 I changed `return i` to be `return i / Float(ts_per_hr)`, where `ts_per_hr` is either (A) 1 when performed on hourly data, or (B) the number of timesteps per hour when performed on subhourly data.